core/trino-main/src/main/java/io/trino/server/protocol/JsonEncodingUtils.java

```diff
@@ -143,10 +160,51 @@ public static void writePagesToJsonGenerator(Consumer<TrinoException> throwableC
         }
     }
 
+    private static String describeBlock(Block block)
```

Comment:
Can we improve the Block#toString implementations to print this instead ?
It would be beneficial for debugging in general